exatomic/nwchem/output.py

```diff
@@ -257,6 +264,167 @@ def parse_basis_set_order(self):
+                mat[i][j] = 0.5 * (mat[i][j] + np.transpose(mat[i][j]))
+        mat = mat.reshape(18,3)
+        df3 = pd.DataFrame(mat, columns=['x', 'y', 'z'])
+        df3['grp1'] = [i for i in range(2) for j in range(9)]
+        df3['grp2'] = [j for i in range(2) for j in range(3) for n in range(3)]
+        df3 = pd.DataFrame(df3.groupby(['grp1','grp2']).apply(lambda x: 
+                                x.unstack().values[:-6]).values.tolist(), 
+                                columns=['xx', 'xy', 'xz', 'yx', 'yy', 'yz', 'zx', 'zy', 'zz'])
+        df['label'] = found_2d.keys()
+        df['label'].replace([_reare, _reombre, _reaim, _reombim], 
+                                ['alpha_re', 'g_re', 'alpha_im', 'g_im'], inplace=True)
+        df['frame'] = np.repeat([0], len(df.index))
+        df3['label'] = np.repeat(list(found_3d.keys()), 3)
+        df3['label'].replace([_redqre, _redqim], ['A_real', 'A_imag'], inplace=True)
+        df3['frame'] = np.repeat([0], len(df3.index))
+        self.roa = pd.concat([df, df3], ignore_index=True)
+
+    def parse_frequency(self):
+        """
+        Parse the :class:`~exatomic.core.atom.Frequency` dataframe.
+
+        Note:
+            This code removes all negative frequencies.
+        """
+        _remeth = "NORMAL MODE EIGENVECTORS IN CARTESIAN COORDINATES"
+        _refreq = "Frequency"
+        _renat = "Atom information"
+
+        found = self.find(_remeth)
+        fnat = self.find(_renat)
+        if not found and not fnat:
+            return
+        # get atom information
+        start = fnat[0][0]+3
+        stop = start
+        while '----' not in self[stop]: stop += 1
+        # we assume that there is only one instance of where _renat is found
+        columns = ['symbol', 'atom', 'x', 'y', 'z', 'mass']
+        atom = self.pandas_dataframe(start, stop, columns)
+        atom['atom'] -= 1
+        nat = len(atom)
+        # find bounds where the calculated frequencies are
+        start = found[0][0]
+        stop = found[1][0]
+        # get the data
+        found = self.find(_refreq, start=start, stop=stop)
+        dfs = []
+        fdx = 0
+        # get frequencies
+        for lno, ln in found:
+            # get the frequency values
+            tmp = ln.split()[1:]
+            freq = np.asarray([float(i) for i in tmp])
+            ## TODO: here we remove all negative frequencies
+            ##       need to find out if this is ok to do
+            # set start and end points for the calculated normal modes
+            staf = lno+start+1
+            stof = lno+start+nat*3+2
+            nm = self.pandas_dataframe(staf, stof, ncol=len(freq)).reset_index(drop=True)
+            # generate boolean array that shows False for negative frequencies
+            neg = [False if f < 0 else True for f in freq]
+            # remove negative frequencies
+            nm.drop(columns=[idx for idx, val in enumerate(neg) if not val], inplace=True)
```

Comment:
Also, when doing boolean checks, it can sometimes be cleaner to allow direct evaluation of the condition, e.g.

```python
neg = [not f < 0 for f in freq]
```

Reply:
I'm not quite sure that I understand your comment @avmarchenko . I have changed line 373 to what @tjduigna suggested.